### PR TITLE
PYIC-4946: Implemented new GA4 library

### PR DESCRIPTION
**Pull request by RogVanGDS**
### What changed

Implemented new GA4 library node module and ripped out the old implementation.

### Why did it change

Moving to the new library to make it as easy as possible for the various pods that make up the One Login journey to upgrade their analytics while having as minimal an impact as possible on the dev teams time and effort.

We need to remove our current GA4 implementation to avoid duplicating logic.

### Issue tracking

- [PYIC-4946](https://govukverify.atlassian.net/browse/PYIC-4946)

### Environment variables or secrets

- [ ] No environment variables or secrets were added or changed

- [ ] Documented in the [README](./blob/main/README.md)
- [ ] Added to deployment repository
- [ ] Added to local startup repository

### Other considerations

- [ ] Update [README](./blob/main/README.md) with any new instructions or tasks

[PYIC-4946]: https://govukverify.atlassian.net/browse/PYIC-4946?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/views/shared/banner.njk [hunk 1]
```diff
@@ -76,5 +76,5 @@
       ],
       hidden: true
     }
-  ],  hidden: true
+  ],  hidden: false
```
Comment (blakeyp) [inline src/views/shared/banner.njk:79]: was this intentionally changed?

File: src/views/shared/ga4/on-page-load-script.njk [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+            .DI
+            .analyticsGa4
+            .pageViewTracker
+            .trackOnPageLoad({
+                language: document.querySelector("html")?.getAttribute("lang"),
+                location: document.location.href?.toLowerCase(),
+                englishPageTitle: '{{params.englishPageTitle}}',
+                referrer: document.referrer?.toLowerCase(),
+                content_id: "",
+                dynamic: '{{params.isPageDynamic}}' ?? false,
```
Comment (blakeyp) [inline src/views/shared/ga4/on-page-load-script.njk]: should this be wrapped in a string or can just pass the bool as is?

File: deploy/template.yaml [hunk 3]
```diff
@@ -168,6 +178,8 @@ Mappings:
       nodeOldSpaceLimit: "1792"
       nodeEnv: "production"
       desiredTaskCount: 2
+      ga4Disabled: true
+      uaDisabled: true
```
Comment (blakeyp) [inline deploy/template.yaml]: Based on https://govukverify.atlassian.net/browse/DCP-3156?focusedCommentId=126209 this should initially be true, false in prod (and integration)?

File: src/views/shared/banner.njk (hunk 1 above)
Comment (RogVanGDS) [reply src/views/shared/banner.njk:79]: Yes, the banner needs to be displayed by default. The logic in the analytics package is to hide it if the user has opted out, not display it if the user has opted in so we want display by default.

At least that is my understanding and the logic is the same for the mobile-app but I suggest we test this before going live

File: src/views/shared/ga4/on-page-load-script.njk (hunk 2 above)
Comment (RogVanGDS) [reply src/views/shared/ga4/on-page-load-script.njk]: We want to be passing in a boolean here

Comment (blakeyp) [reply src/views/shared/ga4/on-page-load-script.njk]: Sorry I mean the `'{{params.isPageDynamic}}'` is wrapped in `''` when it shouldn't be?

Comment (RogVanGDS) [reply src/views/shared/ga4/on-page-load-script.njk]: Ah of course 🤦

File: src/views/shared/ga4/on-page-load-script.njk [hunk 4]
```diff
@@ -0,0 +1,22 @@
...
+            .DI
+            .analyticsGa4
+            .pageViewTracker
+            .trackOnPageLoad({
+                language: document.querySelector("html")?.getAttribute("lang"),
+                location: document.location.href?.toLowerCase(),
+                englishPageTitle: '{{ params.englishPageTitle }}',
+                referrer: document.referrer?.toLowerCase(),
+                content_id: "",
+                dynamic: '{{ params.isPageDynamic }}' === 'true',
```
Comment (RogVanGDS) [inline src/views/shared/ga4/on-page-load-script.njk:13]: Turns out it does need to be wrapped in quotation marks after all 🤷

**Review by shivanshuit914 — APPROVED** [review]

**Review by blakeyp — APPROVED** [review]

**Pull request merged**